### PR TITLE
chore: bump opendal to v0.45.1 for connector/icelake

### PR DESCRIPTION
**Pull request by xiangjinwu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Supplement to #15744.

Note that icelake specifies `>=0.40`, causing dependabot to automatically bump it to `0.45`. But our connector requires `0.44`, leading to all dependabot PRs blocked.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by xxchan — APPROVED** [review]

File: src/connector/src/source/filesystem/opendal_source/opendal_reader.rs [hunk 1]
```diff
@@ -126,6 +128,7 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         let reader = op
             .reader_with(&object_name)
             .range(split.offset as u64..)
+            .into_future() // Unlike `rustc`, `try_stream` seems require manual `into_future`.
```
Comment (BugenZhao) [inline src/connector/src/source/filesystem/opendal_source/opendal_reader.rs:131]: What does `rustc` refer to? 👀

Comment (xiangjinwu) [reply src/connector/src/source/filesystem/opendal_source/opendal_reader.rs:131]: It refers to the normal `.await` [desugaring](https://doc.rust-lang.org/std/future/trait.IntoFuture.html#await-desugaring) performed by `rustc`, in contrast to how the proc macro `try_stream` transforms the `.await` syntax.

Comment (BugenZhao) [reply src/connector/src/source/filesystem/opendal_source/opendal_reader.rs:131]: This is interesting. Heard about `IntoFuture` a long time ago and finally meet one in practice.

Comment (xiangjinwu) [reply src/connector/src/source/filesystem/opendal_source/opendal_reader.rs:131]: `opendal::Operator::reader_with` returns:
* (v0.44.2) `FutureReader` which implements `Future`
* (v0.45.1) `FutureRead` which implements `IntoFuture`

**Pull request merged**